streams/src/test/java/org/apache/kafka/streams/processor/internals/metrics/StreamsMetricsImplTest.java

```diff
@@ -230,6 +244,111 @@ public void testTotalMetricDoesntDecrease() {
+
+        addTotalMetricToSensor(sensor, group, tags, metricNamePrefix, description1);
+
+        verifySumMetric(metrics, sensor, description1, time, 0);
+        assertThat(metrics.metrics().size(), equalTo(1 + 1)); // one metric is added automatically in the constructor of Metrics
+    }
+
+    @Test
+    public void shouldAddAmountRate() {
+        final MockTime time = new MockTime(0);
+
+        addAmountRateMetricToSensor(sensor, group, tags, metricNamePrefix, description1);
+
+        verifyAmountRateMetric(metrics, sensor, description1, time);
+        assertThat(metrics.metrics().size(), equalTo(1 + 1)); // one metric is added automatically in the constructor of Metrics
+    }
+
+    @Test
+    public void shouldAddValue() {
+        final MockTime time = new MockTime(0);
+
+        addValueMetricToSensor(sensor, group, tags, metricNamePrefix, description1);
+
+        final KafkaMetric ratioMetric = metrics.metric(new MetricName(metricNamePrefix, group, description1, tags));
+        assertThat(ratioMetric, is(notNullValue()));
+        final MetricConfig metricConfig = new MetricConfig();
+        final double value1 = 42.0;
+        sensor.record(value1);
+        assertThat(ratioMetric.measurable().measure(metricConfig, time.milliseconds()), equalTo(42.0));
+        final double value2 = 18.0;
+        sensor.record(value2);
+        assertThat(ratioMetric.measurable().measure(metricConfig, time.milliseconds()), equalTo(18.0));
+        assertThat(metrics.metrics().size(), equalTo(1 + 1)); // one metric is added automatically in the constructor of Metrics
+    }
+
+    @Test
+    public void shouldAddAvgAndTotalMetricsToSensor() {
+        final MockTime time = new MockTime(0);
+
+        addAvgAndTotalMetricsToSensor(sensor, group, tags, metricNamePrefix, description1, description2);
+
+        verifyAvgMetric(metrics, sensor, description1, time);
+        verifySumMetric(metrics, sensor, description2, time, 60.0);
+        assertThat(metrics.metrics().size(), equalTo(2 + 1)); // one metric is added automatically in the constructor of Metrics
+    }
+
+    private void verifySumMetric(final Metrics metrics,
+                                 final Sensor sensor,
+                                 final String description,
+                                 final MockTime time,
+                                 final double initialValue) {
+        final KafkaMetric metric = metrics
+            .metric(new MetricName(metricNamePrefix + "-total", group, description, tags));
+        assertThat(metric, is(notNullValue()));
+        assertThat(metric.metricName().description(), equalTo(description));
+        final double value1 = 42.0;
+        sensor.record(value1, time.milliseconds());
+        final double value2 = 18.0;
+        sensor.record(value2, time.milliseconds());
+        assertThat(metric.measurable().measure(new MetricConfig(), time.milliseconds()),
+                   equalTo(initialValue + value1 + value2));
```

Comment:
Where does `initialValue` come from? For example: `shouldAddAvgAndTotalMetricsToSensor` add the sensor and calls `verifySumMetric` -- why is the initial sum not zero?